### PR TITLE
Add image scale option

### PR DESCRIPTION
**Pull request by jodur**
With this change it is possible to scale the received image from a camera entity with a factor (ratio is preserved)

**Note:**  Scale is done before transmitting to deepstack.

Advantages:
- Smaller snapshot file
- Faster processing by deepstack

### CODE REVIEW

**Comment by robmarkcole** [comment]
@jodur have you tried https://www.home-assistant.io/integrations/proxy/ ? This allows downsizing images

Deepstack resizes the images before processing them. This integration saves the full sized image with annotations

**Comment by jodur** [comment]
Yes I am aware of the proxy component. It is a overkill and a loss of unnecessary resources to use the proxy camera, just for rescaling a single snapshot. Every frame from the original cam will be rescaled when the stream from the proxy is accessed. Also an extra delay will be introduced with the proxy for the image processing in relation to realtime motion signals.
The scale property is optional in this case and the extra code for scaling is only processed when a scaling factor different then 1.0 is used. The annotations will also work on the scaled images, so the user “sees”no difference.

**Comment by robmarkcole** [comment]
@jodur RE faster processing did you perform a benchmark measurement to estimate the speed improvement? What % improvement is it?

RE PR can you add the new arg to the README and also perform black formatting, thanks

File: custom_components/deepstack_object/image_processing.py [hunk 1]
```diff
@@ -291,24 +296,40 @@ def __init__(
...
         self._save_file_folder = save_file_folder
         self._save_timestamped_file = save_timestamped_file
+        self._image=None
 
     def process_image(self, image):
         """Process an image."""
-        self._image_width, self._image_height = Image.open(
-            io.BytesIO(bytearray(image))
-        ).size
+        _LOGGER.debug((f"Open image"))
```
Comment (robmarkcole) [inline custom_components/deepstack_object/image_processing.py:310]: "Open image" - can you add to this?

File: custom_components/deepstack_object/image_processing.py [hunk 2]
```diff
@@ -291,24 +296,40 @@ def __init__(
...
-        ).size
+        _LOGGER.debug((f"Open image")) 
+        self._image= Image.open(io.BytesIO(bytearray(image))
+        )
+        
+        self._image_width, self._image_height = self._image.size
+        
+        #resize image if different then default
+        if self._scale!=DEAULT_SCALE:
+            try:
```
Comment (robmarkcole) [inline custom_components/deepstack_object/image_processing.py:318]: How often does this fail?

**Comment by jodur** [comment]
Updated the PR, with your remarks

**Comment by jodur** [comment]
I changed the PR with your remarks:
- deleted exception handeling for rescaling
- code formatting using black

File: custom_components/deepstack_object/image_processing.py [hunk 3]
```diff
@@ -291,19 +298,35 @@ def __init__(
...
         self._save_timestamped_file = save_timestamped_file
+        self._image = None
 
     def process_image(self, image):
         """Process an image."""
-        self._image_width, self._image_height = Image.open(
-            io.BytesIO(bytearray(image))
-        ).size
+        self._image = Image.open(io.BytesIO(bytearray(image)))
+        _LOGGER.debug((f"Open image"))
```
Comment (robmarkcole) [inline custom_components/deepstack_object/image_processing.py:313]: Please make this useful or remove

**Review by robmarkcole — APPROVED** [review]

**Comment by robmarkcole** [comment]
LGTM, nice work

**Pull request merged**